common/djangoapps/django_comment_common/utils.py

```diff
@@ -124,12 +124,13 @@ def set_course_discussion_settings(course_key, **kwargs):
     Returns:
         A CourseDiscussionSettings object.
     """
-    fields = {'division_scheme': str, 'always_divide_inline_discussions': bool, 'divided_discussions': list}
+    fields = {'division_scheme': basestring, 'always_divide_inline_discussions': bool, 'divided_discussions': list}
     course_discussion_settings = get_course_discussion_settings(course_key)
     for field, field_type in fields.items():
         if field in kwargs:
             if not isinstance(kwargs[field], field_type):
                 raise ValueError("Incorrect field type for `{}`. Type must be `{}`".format(field, field_type.__name__))
             setattr(course_discussion_settings, field, kwargs[field])
+    if kwargs:
```

Comment:
I added it since it seemed like a bad idea to make a database call if nothing was modified, however it looks like the discussion/views.py course_discussion_settings_handler method has it's own null check for whether or not the kwargs exists. It could be removed in one of the two places but both is unnecessary.